From 6422b5c0dbd7b5602f449ccec702266cdde2b34e Mon Sep 17 00:00:00 2001
From: He Chuyue <hechuyue@wxiat.com>
Date: Thu, 19 May 2022 09:06:17 +0800
Subject: [PATCH] sw64: fix some structs related to pt_regs
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 6422b5c0dbd7b5602f449ccec702266cdde2b34e
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5GF7A

--------------------------------

Because of previous modifications to struct pt_regs, this patch
fixes struct pt_regs_offset, dbg_reg_def and perf_event_sw64_regs
to keep them consistent.

Signed-off-by: He Chuyue <hechuyue@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/uapi/asm/perf_regs.h |  7 +++++++
 arch/sw_64/kernel/kgdb.c                | 14 +++++++-------
 arch/sw_64/kernel/ptrace.c              |  7 +++++++
 3 files changed, 21 insertions(+), 7 deletions(-)

diff --git a/arch/sw_64/include/uapi/asm/perf_regs.h b/arch/sw_64/include/uapi/asm/perf_regs.h
index 426ae642fcc8..1378a7397951 100644
--- a/arch/sw_64/include/uapi/asm/perf_regs.h
+++ b/arch/sw_64/include/uapi/asm/perf_regs.h
@@ -13,6 +13,13 @@ enum perf_event_sw64_regs {
 	PERF_REG_SW64_R6,
 	PERF_REG_SW64_R7,
 	PERF_REG_SW64_R8,
+	PERF_REG_SW64_R9,
+	PERF_REG_SW64_R10,
+	PERF_REG_SW64_R11,
+	PERF_REG_SW64_R12,
+	PERF_REG_SW64_R13,
+	PERF_REG_SW64_R14,
+	PERF_REG_SW64_R15,
 	PERF_REG_SW64_R19,
 	PERF_REG_SW64_R20,
 	PERF_REG_SW64_R21,
diff --git a/arch/sw_64/kernel/kgdb.c b/arch/sw_64/kernel/kgdb.c
index 491f287eede9..ac2f397f1609 100644
--- a/arch/sw_64/kernel/kgdb.c
+++ b/arch/sw_64/kernel/kgdb.c
@@ -34,13 +34,13 @@ struct dbg_reg_def_t dbg_reg_def[DBG_MAX_REG_NUM] = {
 	{ "r7", 8, offsetof(struct pt_regs, r7)},
 	{ "r8", 8, offsetof(struct pt_regs, r8)},
 
-	{ "r9",  8, -1 },
-	{ "r10", 8, -1 },
-	{ "r11", 8, -1 },
-	{ "r12", 8, -1 },
-	{ "r13", 8, -1 },
-	{ "r14", 8, -1 },
-	{ "r15", 8, -1 },
+	{ "r9",  8, offsetof(struct pt_regs, r9)},
+	{ "r10", 8, offsetof(struct pt_regs, r10)},
+	{ "r11", 8, offsetof(struct pt_regs, r11)},
+	{ "r12", 8, offsetof(struct pt_regs, r12)},
+	{ "r13", 8, offsetof(struct pt_regs, r13)},
+	{ "r14", 8, offsetof(struct pt_regs, r14)},
+	{ "r15", 8, offsetof(struct pt_regs, r15)},
 
 	{ "r16", 8, offsetof(struct pt_regs, r16)},
 	{ "r17", 8, offsetof(struct pt_regs, r17)},
diff --git a/arch/sw_64/kernel/ptrace.c b/arch/sw_64/kernel/ptrace.c
index ce41d89a54cb..94809804e23e 100644
--- a/arch/sw_64/kernel/ptrace.c
+++ b/arch/sw_64/kernel/ptrace.c
@@ -619,6 +619,13 @@ static const struct pt_regs_offset regoffset_table[] = {
 	REG_OFFSET_NAME(r6),
 	REG_OFFSET_NAME(r7),
 	REG_OFFSET_NAME(r8),
+	REG_OFFSET_NAME(r9),
+	REG_OFFSET_NAME(r10),
+	REG_OFFSET_NAME(r11),
+	REG_OFFSET_NAME(r12),
+	REG_OFFSET_NAME(r13),
+	REG_OFFSET_NAME(r14),
+	REG_OFFSET_NAME(r15),
 	REG_OFFSET_NAME(r19),
 	REG_OFFSET_NAME(r20),
 	REG_OFFSET_NAME(r21),
-- 
2.33.0

